Sources/SwiftKuery/Connection.swift

```diff
@@ -14,64 +14,111 @@
  limitations under the License.
  */
 
+import Dispatch
+
 // MARK: Connection protocol
 
 /// Defines the protocol which all database plugins must implement.
-public protocol Connection {
-  
+public protocol Connection : class {
+
     /// The `QueryBuilder` with connection specific substitutions.
     var queryBuilder: QueryBuilder { get }
     
     /// Establish a connection with the database.
     ///
     /// - Parameter onCompletion: The function to be called when the connection is established.
-    func connect(onCompletion: (QueryError?) -> ())
+    func connect(onCompletion: @escaping (QueryError?) -> ())
+
+    /// Establish a connection with the database.
+    ///
+    /// - Returns: A QueryError if the connection cannot connect, otherwise nil
+    func connectSync() -> QueryError?
```

Comment:
Why do we need to provide all these `...Sync()` versions too? Can't we just be async?

Reply:
I will remove all the sync calls other than connectSync.